developer_manual/digging_deeper/files-metadata.rst

```diff
@@ -0,0 +1,352 @@
+    use OCP\FilesMetadata\Event\MetadataLiveEvent;
+
+    class UpdateFilesMetadata implements IEventListener {
+        public function __construct() {
+        }
+
+        public function handle(Event $event): void {
+            if (!($event instanceof MetadataLiveEvent) &&
+                !($event instanceof MetadataBackgroundEvent)) {
+                return;
+            }
+
+            $node = $event->getNode();
+
+            // my-first-meta is light enough
+            $metadata = $event->getMetadata();
+            $metadata->setString('my-first-meta', 'yes');
+
+            if ($event instanceof MetadataLiveEvent) {
+                $event->requestBackgroundJob();
+                return;
+            }
+
+            // my-second-meta is too heavy and should be run on a background job
+            $metadata->setInt('my-second-meta', 1234, true);
+        }
+    }
+
+
+Read metadata using occ command
+-------------------------------
+
+Stored metadata related to a file can be obtained from a console, using the ``occ`` command:
+
+.. code-block:: console
+
+    $ ./occ metadata:get 1742
+    {
+        "my-first-meta": {
+            "value": "yes",
+            "type": "string",
+            "indexed": false
+        },
+        "my-second-meta": {
+            "value": 1234,
+            "type": "int",
+            "indexed": true
+        }
+    }
+
+.. note::
+    The generation process can also be initiated, using the ``--refresh`` option. Please note that the file owner need to be specified in the command.
+    When refreshing metadata from the console, both ``MetadataLiveEvent`` and ``MetadataBackgroundEvent`` are triggered, without waiting for the next tick of crontab
+
+
+Update metadata using PROPPATCH
+-------------------------------
+
+Using Webdav request, a client can create or update metadata about a file:
+
+.. code-block:: console
+
+    curl 'https://cloud.example.net/remote.php/dav/files/test/document.txt' \
+        --user test:test \
+        --request PROPPATCH \
+        --data '<?xml version="1.0" encoding="UTF-8"?>
+            <d:propertyupdate xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
+                <d:set>
+                    <d:prop>
+                        <nc:metadata-myapp-test>123</nc:metadata-myapp-test>
+                    </d:prop>
+                </d:set>
+            </d:propertyupdate>'
+
+This will return a result like
+
+.. code-block:: xml
+
+    <?xml version="1.0"?>
+    <d:multistatus xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
+        <d:response>
+            <d:href>/remote.php/dav/files/test/document.txt</d:href>
+            <d:propstat>
+                <d:prop>
+                    <nc:metadata-myapp-test/>
+                </d:prop>
+                <d:status>HTTP/1.1 200 OK</d:status>
+            </d:propstat>
+        </d:response>
+    </d:multistatus>
+
```

Comment:
Note: When getting or updating a metadata through WebDAV, metadata are prefixed by `metadata-`. Which means that, in this case, the metadata name is `myapp-test`.